### PR TITLE
Create weight_conversion.py

### PR DESCRIPTION
**Pull request by AnubhavSolanki**
### **Describe your change:**

Added a Weight Conversion Algorithm.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by xcodz-dot** [comment]
* `pip install pre-commit`
* `pre-commit run -a`

**Comment by AnubhavSolanki** [comment]
@xcodz-dot Thank you for your response so what should I do now as all checks have already passed.

**Comment by xcodz-dot** [comment]
@dhruvmanila check this out

**Comment by xcodz-dot** [comment]
@AnubhavSolanki, You can mention maintainers if you feel like your work is done

File: conversions/weight_conversion.py
Comment (xcodz-dot) [inline conversions/weight_conversion.py:24]:
```suggestion
    >>> weight_conversion("kilogram","kilogram",5)
```

File: conversions/weight_conversion.py
Comment (xcodz-dot) [inline conversions/weight_conversion.py:26]:
```suggestion
    >>> weight_conversion("kilogram","carrat",5)
```

File: conversions/weight_conversion.py
Comment (xcodz-dot) [inline conversions/weight_conversion.py:28]:
```suggestion
    >>> weight_conversion("ounce","short-ton",3)
```

File: conversions/weight_conversion.py
Comment (xcodz-dot) [inline conversions/weight_conversion.py:6]:
```suggestion
def weight_conversion(from_type: str, to_type: str, value: float) -> any:
```
Python naming convention

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
Naming Convention

**Review by xcodz-dot — APPROVED** [review]
Great

**Comment by AnubhavSolanki** [comment]
so now can we merge it?

**Comment by xcodz-dot** [comment]
Yeah wait for @dhruvmanila, only members can merge (I am just a guy helping people)

**Comment by AnubhavSolanki** [comment]
> Yeah wait for @dhruvmanila, only members can merge (I am just a guy helping people)

Oh GOT IT... THANKS BTW

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:6]:
```suggestion
def weight_conversion(from_type: str, to_type: str, value: float) -> float:
```
This should probably be float. Please make sure by running:
```
$ python -m pip install mypy
$ mypy conversions/weight_conversion.py
```

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:21]: Please include this link in the module-level docstring as well.

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:28]: Please include tests for other conversions as well.

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:31]: This is a global constant. So, put this in the global scope with all uppercase letters like so: `KILOGRAM_CHART`

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:44]: Same as above.

Also, what is the difference between this (`weight_type_chart`) and `kilogram_chart` ?

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:2]:
```suggestion
Conversion of weight units.
```
Please provide the usage of the function, like how would an unknown person use this.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

**Comment by xcodz-dot** [comment]
@AnubhavSolanki, you can directly use the commit button near suggestions if you do not like manual working

File: conversions/weight_conversion.py
Comment (AnubhavSolanki) [reply conversions/weight_conversion.py:44]: It calculates the result through the multiplication of both charts.

File: conversions/weight_conversion.py
Comment (AnubhavSolanki) [reply conversions/weight_conversion.py:21]: I don't get you can u explain (module-level docstring)?

Comment (xcodz-dot) [reply conversions/weight_conversion.py]: Module level docstring is
```python
"""
This is a module level docstring
"""
class someclass():
    """
    This is class level docstring
    """
```

**Comment by xcodz-dot** [comment]
@dhruvmanila, check this if it meets your requirements

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:270]:
```suggestion
    return value * KILOGRAM_CHART[to_type] * WEIGHT_TYPE_CHART[from_type]
```
`dict.get` is used if there's a chance that the key does not exist in the dictionary and you want some default to be returned. In this case, it will return `None` which gets `mypy` to throw an error. As your function parameters are compulsory, there's no change that the default value will be used.

Also, I suggest to put this in a `try ... except ...` block as if any of the `from_type` or `to_type` does not exist, then it will blow up.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
Just a small tweak to satisfy `mypy` and we will be good to go, awesome work so far.

File: conversions/weight_conversion.py
Comment (xcodz-dot) [reply conversions/weight_conversion.py:270]: thats good, will work, also will raise KeyError if invalid information is passed. After this change it seem good to go

**Review by BryanLaura — APPROVED** [review]

**Comment by AnubhavSolanki** [comment]
@dhruvmanila, check this if it's ok.

**Review by BryanLaura — APPROVED** [review]

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:280]:
```suggestion
    except KeyError as err:
        print(
            f"Invalid 'from_type' or 'to_type' value: {err}\n"
            f"Supported values are: {', '.join(WEIGHT_TYPE_CHART)}"
        )
```
Multiplying an `int` to a `float` does not raise any exception. As you're accessing a key in a dictionary, if the key is not present it will raise a `KeyError`.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: conversions/weight_conversion.py
Comment (dhruvmanila) [inline conversions/weight_conversion.py:280]: The reason for this change is so that the static type checker `mypy` does not complain that the function had no return statement.

If we did the `try ... except ...` pattern, when the `try` block raises an exception, we go to the `except` block where we would normally just print a message. This would mean that the function returns `None` implicitly, thus the `mypy` error.

Hope you understand the choice, if not please feel free to ask any question.

**Review by dhruvmanila — APPROVED** [review]
Awesome work! Thank you for your contribution.

File: conversions/weight_conversion.py
Comment (AnubhavSolanki) [reply conversions/weight_conversion.py:280]: thank you got your point.. :)

**Pull request merged**
